### ISSUE TITLE
require(require.resolve(...)) ?

### ISSUE THREAD

**Issue by temsa** [pinojs/pino#216]
I am wondering why pino has this code in `lib/tools.js`:

```js
var levels = require(require.resolve('./levels'))
var serializers = require(require.resolve('./serializers'))
var time = require(require.resolve('./time'))
```

it prevents webpack to load the node version in the browser

**Comment by davidmarkclements** [comment]
I've no idea - afaik it's not necessary at all we can just do

```js
var levels = require('./levels'))
var serializers = require('./serializers')
var time = require('./time')
```

would you like to send a PR (unless @jsumners knows why not to, which me may do)

on a side rant - I really don't know why webpack has to scan all JS files that aren't relevant to it

on a productive note - we should probably make a webpack build test - PR's welcome

**Comment by mcdonnelldean** [comment]
@mcollina This is the issue we where discussing FYI

**Comment by temsa** [comment]
I'll make a quick PR then :)

**Comment by jsumners** [comment]
Because, unless I'm missing it in the source code, `require()` doesn't do a full resolution of relative paths before using them as the cache key. This matters because `lib/levels` is also included in `pino.js`. By giving `require` the full path for the key, we avoid recreating the `lsCache` and the standard level verification caches -- https://github.com/pinojs/pino/blob/1048c181d0513d62c3472f059a0e7eb47a4c4b4c/lib/levels.js#L20-L44. This is important since these are needed to keep track of custom levels.

I'd welcoming being proved wrong.

**Comment by mcollina** [comment]
Fixed in #217 .
